### PR TITLE
Feature #13601 PRODPROCESSLINE : duration

### PR DESCRIPTION
**Pull request by ego-axelor**

### CODE REVIEW

File: axelor-production/src/main/java/com/axelor/apps/production/service/OperationOrderWorkflowService.java [hunk 1]
```diff
@@ -457,24 +456,19 @@ public LocalDateTime computePlannedEndDateT(OperationOrder operationOrder)
...
     ProdProcessLine prodProcessLine = operationOrder.getProdProcessLine();
     WorkCenter workCenter = prodProcessLine.getWorkCenter();
 
     long duration = 0;
+    
+    BigDecimal maxCapacityPerCycle = prodProcessLine.getMaxCapacityPerCycle();
+
+    BigDecimal nbCycles;
+    if (maxCapacityPerCycle.compareTo(BigDecimal.ZERO) == 0) {
+  	  nbCycles = new BigDecimal(1);
```
Comment (gdu-axelor) [inline axelor-production/src/main/java/com/axelor/apps/production/service/OperationOrderWorkflowService.java]: BigDecimal.ONE

File: axelor-production/src/main/java/com/axelor/apps/production/service/OperationOrderWorkflowService.java [hunk 2]
```diff
@@ -489,44 +483,13 @@ public long computeMachineDuration(OperationOrder operationOrder, BigDecimal qty
...
-        duration += qty.multiply(durationPerCycle).longValue();
-      } else {
-        duration +=
-            (qty.divide(maxCapacityPerCycle, RoundingMode.HALF_UP))
-                .multiply(durationPerCycle)
-                .longValue();
-      }
-
       duration += machine.getEndingDuration();
+      duration += nbCycles.add(new BigDecimal(-1)).multiply(new BigDecimal(machine.getSetupDuration())).longValue();
```
Comment (gdu-axelor) [inline axelor-production/src/main/java/com/axelor/apps/production/service/OperationOrderWorkflowService.java]: better to use subtract method

File: axelor-production/src/main/java/com/axelor/apps/production/service/OperationOrderWorkflowService.java (hunk 1 above)
Comment (ego-axelor) [reply axelor-production/src/main/java/com/axelor/apps/production/service/OperationOrderWorkflowService.java]: It was in the original code. I guessed that if you put 0 as the maxCapacity it should mean that you can do one product at a time so the number of cycles is the qty (then I think that there is a mistake nbCycles should be equal to qty and not to BigDecimal(1)).

File: axelor-production/src/main/java/com/axelor/apps/production/service/OperationOrderWorkflowService.java (hunk 2 above)
Comment (ego-axelor) [reply axelor-production/src/main/java/com/axelor/apps/production/service/OperationOrderWorkflowService.java]: done

File: axelor-production/src/main/java/com/axelor/apps/production/service/OperationOrderWorkflowService.java (hunk 1 above)
Comment (gdu-axelor) [reply axelor-production/src/main/java/com/axelor/apps/production/service/OperationOrderWorkflowService.java]: No, I means that newBigDecimal(1) can be replaced per BigDecimal.ONE

Comment (ego-axelor) [reply axelor-production/src/main/java/com/axelor/apps/production/service/OperationOrderWorkflowService.java]: Ok got it so it is not necessary now

**Pull request merged**